app/controllers/chat_channel_memberships_controller.rb

```diff
@@ -11,41 +16,88 @@ def find_by_chat_channel_id
     )
   end
 
+  def edit
+    @membership = ChatChannelMembership.find(params[:id])
+    @channel = @membership.chat_channel
+    authorize @membership
+  end
+
   def create
-    @chat_channel = ChatChannel.find(permitted_params[:chat_channel_id])
+    membership_params = params[:chat_channel_membership]
+    @chat_channel = ChatChannel.find(membership_params[:chat_channel_id])
     authorize @chat_channel, :update?
-    ChatChannelMembership.create(
-      user_id: permitted_params[:user_id],
-      chat_channel_id: @chat_channel.id,
-      status: "pending",
-    )
+    usernames = membership_params[:invitation_usernames].split(",")
```

Comment:
it can be moved to ChatChannel file, by updating add_user method to handle the ChatChannelMembership creation for all cases, or a separate service can be created for ChatChannelMembership creation